lib/parsers/acorn-extensions.js

```diff
@@ -15,154 +15,180 @@ exports.enableTolerance = function (parser) {
-    // export def from '...'
-    const specifier = this.startNode();
-    specifier.exported = this.parseIdent(true);
-    node.specifiers = [
-      this.finishNode(specifier, "ExportDefaultSpecifier")
-    ];
-    if (this.type === tt.comma &&
-        peekNextType(this) === tt.star) {
-      // export def, * as ns from '...'
-      this.expect(tt.comma);
-      const specifier = this.startNode();
-      this.expect(tt.star);
-      this.expectContextual("as");
-      specifier.exported = this.parseIdent(true);
-      node.specifiers.push(
-        this.finishNode(specifier, "ExportNamespaceSpecifier")
-      );
-    } else {
-      // export def, { x, y as z } from '...'
-      this.parseExportSpecifiersMaybe(node);
-    }
-    this.parseExportFrom(node, exports);
-  } else if (this.eat(tt._default)) {
-    // export default ...
-    exports.default = true;
-    let isAsync;
-    if (this.type === tt._function || (isAsync = this.isAsyncFunction())) {
-      let fNode = this.startNode();
-      this.next();
-      if (isAsync) this.next();
-      node.declaration = this.parseFunction(fNode, "nullableID", false, isAsync);
-    } else if (this.type === tt._class) {
-      let cNode = this.startNode();
-      node.declaration = this.parseClass(cNode, "nullableID");
-    } else {
-      node.declaration = this.parseMaybeAssign();
-      this.semicolon();
-    }
-    return this.finishNode(node, "ExportDefaultDeclaration");
-  } else if (this.shouldParseExportDeclaration()) {
-    // export var|const|let|function|class ...
-    node.declaration = this.parseStatement(true);
-    if (node.declaration.type === "VariableDeclaration") {
-      this.checkVariableExport(exports, node.declaration.declarations);
-    } else {
-      exports[node.declaration.id.name] = true;
+    return parseExportNamespace(this, node, exports);
+  }
+  if (isExportDefaultSpecifier(this)) {
+    return parseExportImportDefault(this, node, exports);
+  }
+  if (this.eat(tt._default)) {
+    return parseExportDefault(this, node, exports);
+  }
+  if (this.shouldParseExportStatement()) {
+    return parseExportDeclaration(this, node, exports);
+  }
+  return parseExportImportSpecifiers(this, node, exports);
+}
+
+function isCommaOrFrom(parser) {
+  return parser.type === tt.comma || parser.isContextual("from");
+}
+
+function isExportDefaultSpecifier(parser) {
+  return parser.type === tt.name &&
+    withLookAhead(parser, 1, isCommaOrFrom);
+}
+
+function parseExportDeclaration(parser, node, exports) {
+  // export var|const|let|function|class ...
+  node.declaration = parser.parseStatement(true);
+  node.source = null;
+  node.specifiers = [];
+
+  if (node.declaration.type === "VariableDeclaration") {
+    parser.checkVariableExport(exports, node.declaration.declarations);
+  } else {
+    exports[node.declaration.id.name] = true;
+  }
+  return parser.finishNode(node, "ExportNamedDeclaration");
+}
+
+function parseExportDefault(parser, node, exports) {
```

Comment:
The convention in Acorn is to include the full AST node type in the function name, if the function is responsible for parsing that node type. So I would call this `parseExportDefaultDeclaration`.